src/arrayMethodSort.js

```diff
@@ -5,7 +5,27 @@
  */
 function applyCustomSort() {
   [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+    for (let i = 0; i < this.length; i++) {
+      if (!compareFunction) {
+        for (let k = i; k < this.length; k++) {
+          if (String(this[k]) < String(this[i])) {
+            this.splice(i, 0, this[k]);
+            this.splice(k + 1, 1);
+          }
+        }
+      }
+
+      if (compareFunction) {
+        for (let j = i; j < this.length; j++) {
+          if (compareFunction(this[i], this[j]) === 1) {
+            this.splice(i, 0, this[j]);
+            this.splice(j + 1, 1);
+          }
+        }
+      }
```

Comment:
This logic is now very similar and duplicated, so you need to create a helper function as the documentation says and use it. And provide this function as default parameter to the method
<img width="805" alt="image" src="https://github.com/mate-academy/js_array-method-sort/assets/64750958/b072e7cd-f4de-4f9c-a737-45a9b339085b">